sql: clean up explain code

RaduBerinde opened the pull request:
Some cleanup of the explain plan code:
 - use a function to encapsulate the logic of how we put values into rows, and
   reuse it for the special distributed/vectorized rows;
 - planToString now uses common logic to emit the nicely formatted tree.

Release note: None

jordanlewis reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>, nice clean up!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/45714#-:-M1an0CdCC2ih-Ts51vG:b-yzw373)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @jordanlewis)



<!-- Sent from Reviewable.io -->

RaduBerinde committed:
sql: clean up explain code

Some cleanup of the explain plan code:
 - use a function to encapsulate the logic of how we put values into rows, and
   reuse it for the special distributed/vectorized rows;
 - planToString now uses common logic to emit the nicely formatted tree.

Release note: None

RaduBerinde commented:
bors r+

RaduBerinde commented:
bors r-

RaduBerinde commented:
bors r+
